synapse/module_api/__init__.py

```diff
@@ -1666,6 +1667,46 @@ async def set_displayname(
             deactivation=deactivation,
         )
 
+    R = TypeVar("R")
+
+    def run_as_background_process(
```

Comment:
It looks like we expose `run_in_background` in a different way that avoids all this. Can we follow the same pattern?

https://github.com/matrix-org/synapse/blob/2e59e97ebd02e93da39e6c90335d3b24ed01217a/synapse/module_api/__init__.py#L158

Reply:
Oh that is easier, thanks for the suggestion!